Pipes - Iuliia - solar_system.rb

```diff
@@ -0,0 +1,140 @@
+class Planet
+  attr_accessor :name, :mass, :distance, :diameter, :orbit
+
+  def initialize(name, mass, distance, diameter, orbit)
+    @name = name
+    @mass = mass.to_f
+    @distance = distance.to_f
+    @diameter = diameter.to_f
+    @orbit = orbit.to_f
+  end
+
+  def info
+    return "Planet: #{@name}\nMass: #{@mass}\nDistance: #{@distance}\nDiameter: #{@diameter}\nOrbit: #{@orbit}"
+  end
+
+end #end of class Planet
+
+class SolarSystem
+
+  attr_accessor :planets, :age
+
+  def initialize(initial_planets, age)
+    @planets = initial_planets
+    @age = age
+  end
+
+  def add_planet(planet)
+    @planets << planet
+  end
+
+  def planet_list
+    result = ""
+    @planets.each do |planet|
+      result +=  "#{planet.info}\n\n"
+    end
+    return result
+  end
+
+  def planet_name_str
+    planet_arr = []
+    i = 0
+    @planets.each do |planet|
+      i += 1
+      planet_arr << "#{i}. #{planet.name}"
+    end
+    return planet_arr.join(", ")
+  end
+
+  def planet_name_by_index(input)
+    if input <= 0 || input > @planets.length
```

Comment:
Note that this method doesn't return the planet name, it returns a `Planet` object. The difference is subtle but important. A better name for this method might be `planet_by_index`.